ocaml/libs/http-lib/http.ml

```diff
@@ -934,81 +934,67 @@ module Url = struct
     scheme_equal a_scheme b_scheme && data_equal a_data b_data
 
   let of_string url =
-    let sub_before c s = String.sub s 0 (String.index s c) in
-    let sub_after c s =
-      let length = String.length s in
-      let start = String.index s c + 1 in
-      String.sub s start (length - start)
-    in
-    let host x =
-      try x |> sub_after '[' |> sub_before ']'
-      with Not_found -> (
-        try (* [<ipv6-literal>]... *)
-            x |> sub_before ':'
-        with Not_found -> (* <hostname|ipv4-literal>:... *)
-                          x
-      )
-    in
-    (* <hostname|ipv4-literal> *)
-    let port x =
-      let port_part =
-        try x |> sub_after ']' |> sub_after ':'
-        with Not_found -> (
-          try (* ...]:port *)
-              x |> sub_after ']'
-          with Not_found -> (
-            try (* ...] *)
-                x |> sub_after ':'
-            with Not_found -> (* ...:port *)
-                              ""
-          )
-        )
+    debug "%s: %s" __FUNCTION__ url ;
+    let fail fmt = Printf.ksprintf failwith fmt in
+    let query = function k, v :: _ -> (k, v) | k, [] -> (k, "") in
```

Comment:
The previous implementation keeps all values for a key, is it significant, though?

Reply:
The URI module can represent a key with multiple values but our representation would have to use duplicate keys - which I think is very questionable and something we should avoid. Currently tests are passing, suggesting we don't use multiple values/keys.